### PR TITLE
@W-17160682 Set SSR env var to 'true' in server-side rendering Jest preset

### PR DESCRIPTION
**Pull request by Shinoni**
Upgrade LWC specific packages to 8.1.2 and set SSR env var to true in the server-side rendering Jest preset

### CODE REVIEW

File: example-ssr/README.md [hunk 1]
```diff
@@ -51,8 +51,8 @@ At present, hydration errors are tracked by monitoring the console.warn event.
 
 You can run server-side and client-side tests independently using the following scripts from `package.json`:
 
--   Server-side: [Server-Side Script](https://github.com/salesforce/lwc-test/blob/e43ba7be692b42ce91c992828547a3749df1a932/package.json#L8)
```
Comment (Shinoni) [inline example-ssr/README.md:54]: Don't worry about the changes in this file. I forgot to rebase.

**Review by lpomerleau — COMMENTED** [review]
Thanks! 

1. Do the component libraries need to upgrade to LWC v8 too?
2. Will the [example-ssr](https://github.com/salesforce/lwc-test/tree/master/example-ssr) sample be updated to use `import.meta.env.SSR` to use this new preset?

**Comment by Shinoni** [comment]
> Thanks!
> 
> 1. Do the component libraries need to upgrade to LWC v8 too?
> 2. Will the [example-ssr](https://github.com/salesforce/lwc-test/tree/master/example-ssr) sample be updated to use `import.meta.env.SSR` to use this new preset?

1. Yes! I know many repos already upgraded the LWC to v8.
2. Yes. Currently, we don't have any logic that relies on the SSR env var but I tested this change in ssr-example. Everything worked as expected.

**Comment by lpomerleau** [comment]
> Currently, we don't have any logic that relies on the SSR env var but I tested this change in ssr-example. Everything worked as expected.

This is great, can you check that change in? Does this repo run the SSR unit tests during CI, so we can ensure everything continues to work?

**Comment by Shinoni** [comment]
> > Currently, we don't have any logic that relies on the SSR env var but I tested this change in ssr-example. Everything worked as expected.
> 
> This is great, can you check that change in? Does this repo run the SSR unit tests during CI, so we can ensure everything continues to work?


The CI isn't running tests from the sample project. I'll request this change from Kishor's team, who are now co-owners of the repo along with Nolan, since I don't have the necessary permissions on CircleCI to make these updates.

**Review by lpomerleau — APPROVED** [review]
🎉

**Pull request merged**